src/agent/testresultreader.ts

```diff
@@ -0,0 +1,337 @@
+import ifm = require('./api/interfaces');
+import trp = require('./testrunpublisher');
+
+var fs = require('fs');
+var path = require("path");
+var xmlreader = require('xmlreader');
+
+export class JUnitResultReader implements trp.IResultReader {
+    public readResults(file: string, runContext: trp.TestRunContext) {
+        
+        var testRun2 : ifm.TestRun2;
+        var contents = fs.readFileSync(file, "utf-8");
```

Comment:
We shouldn't use Sync methods during job execution - it's ties up the event loop.  All I/O should use async.  The only exception we have to this is in config code which is sync by it's nature.

If you make this async, then the readResults function needs to be async with callback